libkbfs/disk_journal.go

```diff
@@ -146,35 +145,9 @@ func (j diskJournal) writeLatestOrdinal(o journalOrdinal) error {
 	return j.writeOrdinal(j.latestPath(), o)
 }
 
-func (j diskJournal) clearOrdinals() error {
-	earliestOrdinal, err := j.readEarliestOrdinal()
-	if err != nil {
-		return err
-	}
-	latestOrdinal, err := j.readLatestOrdinal()
-	if err != nil {
-		return err
-	}
-
-	err = ioutil.Remove(j.earliestPath())
-	if err != nil {
-		return err
-	}
-	err = ioutil.Remove(j.latestPath())
-	if err != nil {
-		return err
-	}
-
-	// Garbage-collect the old entries.  TODO: we'll eventually need a
-	// sweeper to clean up entries left behind if we crash right here.
-	for ordinal := earliestOrdinal; ordinal <= latestOrdinal; ordinal++ {
-		p := j.journalEntryPath(ordinal)
-		err = ioutil.Remove(p)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
+// clear completely removes the journal directory.
+func (j diskJournal) clear() error {
+	return ioutil.RemoveAll(j.dir)
```

Comment:
Renaming the journal is a bit more complex as that introduces the question of cleaning up those tempdirs later. But I think removing the ordinal files first makes this no worse than what we have now, and I added a TODO to read the ordinals in memory on startup, and also to treat a single missing ordinal as an empty journal, which would then make clear() "atomic".